pkg/cmd/roachtest/cluster.go

```diff
@@ -469,6 +469,16 @@ func (c *cluster) RunL(ctx context.Context, l *logger, node int, args ...string)
 	}
 }
 
+func (c *cluster) InitEnterprise(ctx context.Context) {
+	licenseKey := os.Getenv("COCKROACH_DEV_LICENSE")
+	if licenseKey == "" {
+		c.t.Fatal("COCKROACH_DEV_LICENSE must be set to run enterprise tests")
+	}
+	sql := "SET CLUSTER SETTING cluster.organization = 'Cockroach Labs - Production Testing';"
+	sql += fmt.Sprintf("SET CLUSTER SETTING enterprise.license = '%s';", licenseKey)
+	c.Run(ctx, 1, fmt.Sprintf("./cockroach sql --insecure -e %q", sql))
```

Comment:
I wonder if this should be done in `roachprod start`. Certainly, it seems worthwhile to set the `cluster.organization` for all `roachprod` clusters. Setting the enterprise license also seems reasonable. Then the tests which require a license could keep a check for the `COCKROACH_DEV_LICENSE` env var like you have above.

PS You should replace the `--insecure` with `--url {pgurl:1}` as that will work on both secure and insecure clusters.